aries_cloudagent/vc/vc_ld/manager.py

```diff
@@ -273,9 +274,17 @@ async def prepare_credential(
         return credential
 
     async def _get_suite_for_credential(
-        self, credential: VerifiableCredential, options: LDProofVCOptions
+        self,
+        document: Union[VerifiableCredential, VerifiablePresentation],
+        options: LDProofVCOptions,
     ) -> LinkedDataProof:
-        issuer_id = credential.issuer_id
+        document_type = document.type[0]
+
+        if document_type == "VerifiableCredential":
+            issuer_id = document.issuer_id
+        if document_type == "VerifiablePresentation":
+            issuer_id = document.holder_id
```

Comment:
Given that we have Python types defined for these objects, I would recommend checking the type with:

```python
if isinstance(document, VerifiableCredential):
```

Reply:
`_get_suite_for_document` seems fair to me!